src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -1287,24 +1175,53 @@ public static CultureInfo GetCultureInfo(string name)
         /// </summary>
         public static CultureInfo GetCultureInfo(string name, string altName)
         {
-            if (name == null)
+            if (name is null)
             {
                 throw new ArgumentNullException(nameof(name));
             }
-            if (altName == null)
+            if (altName is null)
             {
                 throw new ArgumentNullException(nameof(altName));
             }
 
-            CultureInfo? retval = GetCultureInfoHelper(-1, name, altName);
-            if (retval == null)
+            name = CultureData.AnsiToLower(name);
+            altName = CultureData.AnsiToLower(altName);
+            string nameAndAltName = name + "\xfffd" + altName;
+            Dictionary<string, CultureInfo> nameTable = CachedCulturesByName;
+            CultureInfo? result;
+
+            lock (nameTable)
+            {
+                if (nameTable.TryGetValue(nameAndAltName, out result))
+                {
+                    return result;
+                }
+            }
+
+            try
+            {
+                result = new CultureInfo(name, altName) { _isReadOnly = true };
+                result.TextInfo.SetReadOnlyState(readOnly: true); // TextInfo object is already created; we need to set it as read only.
+            }
+            catch (ArgumentException)
+            {
+                throw new CultureNotFoundException("name or altName", SR.Format(SR.Argument_OneOfCulturesNotSupported, name, altName));
+            }
+
+            lock (nameTable)
             {
-                throw new CultureNotFoundException("name or altName",
-                                        SR.Format(SR.Argument_OneOfCulturesNotSupported, name, altName));
+                nameTable[nameAndAltName] = result;
             }
-            return retval;
+
+            return result;
         }
 
+        private static Dictionary<string, CultureInfo> CachedCulturesByName =>
+            LazyInitializer.EnsureInitialized(ref s_cachedCulturesByName, () => new Dictionary<string, CultureInfo>());
```

Comment:
`LazyInitializer` is not cheap. It creates extra method, delegate wrapping that method, etc. Most places in CoreLib use a direct `Interlocked.CompareExchange`.

Reply:
Ok.